### PR TITLE
feat(xslt): context for function-call

### PR DESCRIPTION
**Pull request by AirQuick**
This pull request allows setting `x:context` with `x:call[@function]` when `/x:description/@run-as` is `external`.

`x:context` with `x:call[@function]` works in the same way as `x:call[@template]`. i.e. If `x:call` is preceded by `x:context`, `@function` is invoked once for each item in `x:context`. Each item in `x:context` is the global context item of each function invocation.

### Further work

Update [Wiki](https://github.com/xspec/xspec/wiki/External-Transformation#global-context-item) "Global Context Item" section.

### CODE REVIEW

File: src/compiler/xslt/compile/compile-scenario.xsl [hunk 1]
```diff
@@ -45,11 +45,11 @@
             </xsl:call-template>
          </xsl:message>
       </xsl:if>
-      <xsl:if test="$context and $call/@function">
+      <xsl:if test="$context and $call/@function and not($is-external)">
          <xsl:message terminate="yes">
             <xsl:call-template name="x:prefix-diag-message">
                <xsl:with-param name="message" as="xs:string">
-                  <xsl:text>Can't set a context and call a function at the same time</xsl:text>
+                  <xsl:text expand-text="yes">Setting a context and calling a function at the same time is supported only when /{$initial-document/x:description => name()} has @run-as='external'.</xsl:text>
```
Comment (galtm) [inline src/compiler/xslt/compile/compile-scenario.xsl:52]: A suggestion for streamlining the message: "Setting a context for calling a function is supported only..."

**Review by galtm — APPROVED** [review]
Looks good.

**Comment by AirQuick** [comment]
> ### Further work
> Update [Wiki](https://github.com/xspec/xspec/wiki/External-Transformation#global-context-item) "Global Context Item" section.

[Done](https://github.com/xspec/xspec/wiki/External-Transformation/_compare/28188391a63c4cfbea354f0a7846f88a873dc6be...846e6d63cc86e37ac2eedcd1d10aa9a2dbb5f280).

File: src/compiler/xslt/compile/compile-scenario.xsl (hunk 1 above)
Comment (AirQuick) [reply src/compiler/xslt/compile/compile-scenario.xsl]: Thanks, @galtm. b445bf9642e62e7e461ea52f81d6e3c74f7b9791 integrates it.

I'm merging this.

**Pull request merged**